app/models/ems_event.rb

```diff
@@ -168,8 +166,8 @@ def self.process_middleware_entities_in_event!(event, _options = {})
       unless klass.nil?
         process_object_in_event!(klass, event, :ems_ref_key => :middleware_ref)
       end
-      event.except!(:middleware_ref, :middleware_type)
     end
+    event.except!(:middleware_ref, :middleware_type)
```

Comment:
Because you want to remove those two entries for all events, not just those processed with an entity link.